aptos-move/framework/aptos-token/sources/token_type.move

```diff
@@ -0,0 +1,161 @@
+// token_type provides a method to tell if a token is NFT, NFT print or a fungible token
+// NFT: only one token is allowed to mint from this TokenData. The maximum in TokenData is 1 and the field is immutable
+//
+// NFT_PRINT: this is a NFT with property_version > 0. Only one token has the same token_id
+// But, the maximum of the TokenData is not 1 or the maximum field is mutable at the same time.
+// Note: if token maximum is immutable and set to be 1, the token is still a NFT if property_version > 0
+//
+// FUNGIBLE: a non-decimal token that can have an amount bigger than 1
+module std::token_type {
+
+    use aptos_token::token::{Self, TokenId};
+    #[test_only]
+    use aptos_framework::account;
+    #[test_only]
+    use std::signer;
+    #[test_only]
+    use aptos_token::token::{create_collection_and_token};
+    #[test_only]
+    use std::string::String;
+    #[test_only]
+    use std::string;
+    #[test_only]
+    use std::bcs;
```

Comment:
maybe move all test imports to  the bottom where the tests are

what problem are we solving by adding this? who asked for it?

how did we define the name "NFT_PRINT"

Should FUNGIBLE be distinguished by those that are "SEMI_FUNGIBLE" and purely "FUNGIBLE"? Is that possible?

Reply:
I think the problem is people list their fungible token to sell on marketplace. It is fine. But we should have a standard to tell people if a token is fungible or not. Otherwise, there will be a lot of confusion. 

I have the NFT_print's definition on top of the module description. Please see if it makes sense.

I think a token is either in fungible or non-fungible state. semi-fungible is not a state of a token. It is just a capability of the token.